client/post-editor/controller.js

```diff
@@ -115,6 +115,7 @@ module.exports = {
 
 		function startEditing( siteId ) {
 			context.store.dispatch( setEditorPostId( postID ) );
```

Comment:
I agree that a more meaningfully named action creator might help avoid accidents. Do you think it would be used in addition to or in place of the existing `setEditorPostId` ? If in addition to, would we treat it as a totally separate action type or could we extend the existing action type with additional details / meta to describe it as being a "new" editing session?

>Also, does this mean, we should drop the current `resetPostEdits` calls (`componentWillUnmount`, `componentWillReceiveProps`, `onSaveSuccess`)? I would say yes, but maybe this would change the behavior a bit.

I think it'd be nice to allow the editor component to not worry about these details. If I were to guess, I suspect we reset upon unmounting as a sort of clean-up task, so the state doesn't hang around after leaving the editor. As long as we properly reset when revisiting the editor, I don't know that there will be any problems. Otherwise we could fire some new `stopEditing` action creator upon unmounting / leaving the editor that has similar effects of resetting edit state.